### PR TITLE
feat(connector): [Tsys] Use `connector_response_reference_id` as reference to the connector

### LINKED ISSUE #2319: [FEATURE]: [Tsys] Use `connector_request_reference_id` as reference to the connector
### :memo: Feature Description 

- We transmit our reference id from Hyperswitch to the connector, enabling merchants to subsequently locate the payment within the connector dashboard.
- Presently, there is a lack of uniformity in this process, as some implementations employ `payment_id` while others utilize `attempt_id`.
- This inconsistency arises from the fact that `payment_id` and `attempt_id` correspond to two distinct entities within our system.

### :hammer: Possible Implementation 

- To mitigate this confusion, we've introduced a new field called `connector_request_reference_id` in our RouterData. 
- This id should replace them as a reference id to the connector for payment request. 
- One might need to have exposure to api docs of the connector for which it is being implemented.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/2052

:bookmark: Note : All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### LINKED ISSUE #2350: [FEATURE]: [Tsys] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by Rutam21**
## Type of Change

- [x] New feature

## Description

The `connector_response_reference_id` parameter has been set for the Tsys Payment Solutions for uniform reference and transaction tracking.

### File Changes

- [x] This PR modifies the Tsys Transformers file.

**Location- router/src/connector/tsys/transformers.rs**

## Motivation and Context

This PR was raised so that it Fixes #2350 !

## How did you test it?

- **I ran the following command, and all the errors were addressed properly, and the build was successful.**

```bash
cargo clippy --all-features
```

build

- The code changes were formatted with the following command to fix styling issues.

```bash
cargo +nightly fmt
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

**Comment by Rutam21** [comment]
@srujanchikke Please review the PR and suggest changes, if any. Thanks.

File: crates/router/src/connector/tsys/transformers.rs [hunk 1]
```diff
@@ -192,12 +192,14 @@ fn get_error_response(
...
-        resource_id: types::ResponseId::ConnectorTransactionId(connector_response.transaction_id),
+        resource_id: types::ResponseId::ConnectorTransactionId(
+            connector_response.transaction_id.clone(),
+        ),
         redirection_data: None,
         mandate_reference: None,
         connector_metadata: None,
         network_txn_id: None,
-        connector_response_reference_id: None,
+        connector_response_reference_id: Some(connector_response.transaction_id.clone()),
```
Comment (prasunna09) [inline crates/router/src/connector/tsys/transformers.rs:202]:
```suggestion
        connector_response_reference_id: Some(connector_response.transaction_id),
```

File: crates/router/src/connector/tsys/transformers.rs [hunk 2]
```diff
@@ -215,7 +217,12 @@ fn get_payments_sync_response(
         mandate_reference: None,
         connector_metadata: None,
         network_txn_id: None,
-        connector_response_reference_id: None,
+        connector_response_reference_id: Some(
+            connector_response
+                .transaction_details
+                .transaction_id
+                .clone(),
+        ),
```
Comment (prasunna09) [inline crates/router/src/connector/tsys/transformers.rs:225]:
```suggestion
        connector_response_reference_id: Some(
            connector_response
                .transaction_details
                .transaction_id,
        ),
```

**Review by prasunna09 — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/tsys/transformers.rs (hunk 2 above)
Comment (Rutam21) [reply crates/router/src/connector/tsys/transformers.rs:225]: Hello @prasunna09 , Changing this piece of code throws an error with `cargo clippy --all-features`
error

File: crates/router/src/connector/tsys/transformers.rs (hunk 1 above)
Comment (Rutam21) [reply crates/router/src/connector/tsys/transformers.rs]: This has been implemented now. Thanks for the suggestion.

File: crates/router/src/connector/tsys/transformers.rs (hunk 2 above)
Comment (Rutam21) [reply crates/router/src/connector/tsys/transformers.rs:225]: We are still using the clone property here so that we don't stumble upon this error.

Comment (prasunna09) [reply crates/router/src/connector/tsys/transformers.rs:225]: My bad! Will approve this!

**Review by prasunna09 — APPROVED** [review]

**Pull request closed without merging**